### DISCUSSION TITLE
XML escaped characters

### DISCUSSION THREAD

**Discussion by link4531** [TomWright/dasel#138] [Q&A] [answered]
if I have a yaml file which contains the following

`my.yml`
```>code
This:
      -that: <% something %>
```
`./dasel -f my.yml -r yaml -w xml > my.xml`
will correctly output the xml file:

`my.xml`
```>code
<This that="<% something %>"/>
```
however if I then turn around and read my.xml, dasel complains about the unescaped '<' character. 

```>code
./dasel -f my.xml -p xml
> error unescaped '<'
```
I assume this is an issue with dasel needing a '\\' character to parse the file properly
I understand that if I would need escape character if I was providing a selector to query. but should this happen when I don't even provide a selector to begin with? 

my work arounds for the most part have been:
convert xml to yaml > add in values with '<' or '>' characters > convert back to xml
this works fine, besides that fact that once done, dasel can no longer read that xml file properly to make additional changes. 
or if the initial xml file I am working with already has a special value containing these characters. 

any suggestions?

**Comment by link4531** [comment, marked as answer]
Resolved. This was an internal issue that was causing my xml files to have non xml structure and not specific to dasel.
xml >> Puppet Module ( no longer xml format) >> dasel 
resolution > apply changes using dasel before external module converts files or changes values .

**Reply by TomWright** [reply]
Sorry I didn't actually see this. I'm glad you found the issue!
